snuba/migrations/policies.py

```diff
@@ -0,0 +1,75 @@
+from abc import ABC, abstractmethod
+
+from snuba.migrations.groups import get_group_loader
+from snuba.migrations.runner import MigrationKey, Runner
+from snuba.migrations.status import Status
+
+
+class MigrationPolicy(ABC):
+    """
+    A MigrationPolicy implements an `is_allowed` method
+    that determines whether or not a migration can be applied
+    from snuba admin.
+
+    A policy can be used by a migration group to determine the
+    level of access a group has to run migrations.
+    """
+
+    @abstractmethod
+    def can_run(self, migration_key: MigrationKey) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    def can_reverse(self, migration_key: MigrationKey) -> bool:
+        raise NotImplementedError
+
+
+class ReadOnlyPolicy(MigrationPolicy):
+    """
+    No migration is allowed to be run or reversed.
+    """
+
+    def can_run(self, migration_key: MigrationKey) -> bool:
+        return False
+
+    def can_reverse(self, migration_key: MigrationKey) -> bool:
+        return False
+
+
+class WriteSafeAndPendingPolicy(MigrationPolicy):
+    """
+    Safe migrations can be run in addition to dangerous migrations that
+    are in a pending state. All backwards migrations are considered
```

Comment:
This docstring was a bit unclear to me:
- The migrations framework uses the word `blocking`, here you also introduce the word `safe` and `dangerous` but it's not really clear what these mean. I would prefer to standardise all terminology on blocking.
- I think it would be clearer to say that any migration in a pending state can be reversed. Your comment implies that both directions are ok for pending migrations when it is not.

Reply:
I can clear up this doc string